tests/conditional_compilation/test_cc.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+""" Test conditional compilation.
+"""
+
+import glob
+import logging
+import os
+import sys
+import numpy as np
+import pytest
+
+from proc_utils import cmd_exec  # pylint: disable=import-error
+from test_utils import get_lib_sizes, infer_tool, make_build, run_infer  # pylint: disable=import-error
+
+
+log = logging.getLogger()
+
+
+@pytest.mark.dependency(name="cc_collect")
+def test_cc_collect(test_id, model, openvino_ref, test_info,
+                    save_session_info, sea_runtool, collector_dir, artifacts):  # pylint: disable=unused-argument
+    """Test conditional compilation statistics collection
+    :param test_info: custom `test_info` field of built-in `request` pytest fixture.
+                      contain a dictionary to store test metadata.
+    """
+    out = artifacts / test_id
+    test_info["test_id"] = test_id
+    # cleanup old data if any
+    prev_result = glob.glob(f"{out}.pid*.csv")
+    for path in prev_result:
+        os.remove(path)
+    # run use case
+    sys_executable = (
+        os.path.join(sys.prefix, "python.exe")
+        if sys.platform == "win32"
+        else os.path.join(sys.prefix, "bin", "python")
+    )
+    return_code, output = cmd_exec(
+        [
+            sys_executable,
+            str(sea_runtool),
+            f"--output={out}",
+            f"--bindir={collector_dir}",
+            "!",
+            sys_executable,
+            infer_tool,
+            f"-m={model}",
+            "-d=CPU",
+            f"-r={out}",
+        ]
+    )
+    out_csv = glob.glob(f"{out}.pid*.csv")
+    test_info["out_csv"] = out_csv
+
+    assert return_code == 0, f"Command exited with non-zero status {return_code}:\n {output}"
+    assert len(out_csv) == 1, f'Multiple or none "{out}.pid*.csv" files'
+
+
+@pytest.mark.dependency(depends=["cc_collect"])
+def test_minimized_pkg(test_id, model, openvino_root_dir, artifacts):  # pylint: disable=unused-argument
+    """Build and install OpenVINO package with collected conditional compilation statistics."""
+    install_prefix = artifacts / test_id / "install_pkg"
+    build_dir = openvino_root_dir / "build_minimized"
+
+    log.info("Building minimized build at %s", build_dir)
+
+    return_code, output = make_build(
+        openvino_root_dir,
+        build_dir,
+        install_prefix,
+        cmake_additional_args=[f"-DSELECTIVE_BUILD_STAT={artifacts / test_id}/*.csv"],
```

Comment:
AFAIK `*.csv` isn't really supported by cmake because it won't support wildcards